### PR TITLE
RHOBS-855: rhelemeter: Read client info/data from request

### PR DESCRIPTION
**Pull request by philipgough**
https://issues.redhat.com/browse/RHOBS-855

### CODE REVIEW

File: pkg/authorize/ssl/ssl.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+// ClientCertInfoAsHeaders is middleware that extracts client information from the request and validates
+// the pre-shared key
+func ClientCertInfoAsHeaders(config ClientCertConfig, logger log.Logger) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		fn := func(w http.ResponseWriter, r *http.Request) {
+			secret := r.Header.Get(config.Config.SecretHeader)
+			if secret == "" {
+				http.Error(w, fmt.Sprintf("secret must be sent in request header %s",
+					config.Config.SecretHeader), http.StatusForbidden)
```
Comment (thibaultmg) [inline pkg/authorize/ssl/ssl.go:48]: Small detail, I wonder wether 401 would be more appropriate as here no credentials are provided?

**Review by thibaultmg — APPROVED** [review]
Awesome!
Reading at the related conversation I see that we do not yet check if the system is still registered by calling Candlepin. I guess this is expected?

File: pkg/authorize/ssl/ssl.go (hunk 1 above)
Comment (philipgough) [reply pkg/authorize/ssl/ssl.go:48]: yeah that probably makes more sense. In the absence of a cert the server seems to return a 403 so I followed that but I think its fine to return 401

File: cmd/rhelemeter-server/main_e2e_test.go [hunk 2]
```diff
@@ -159,6 +159,166 @@ func TestServerRHEL(t *testing.T) {
 	}
 }
 
+func TestServerRhelWithClientInfoFromHeaders(t *testing.T) {
+	defer goleak.VerifyNone(t)
+
+	receiveServer := httptest.NewServer(mockedReceiver(t))
+	defer receiveServer.Close()
```
Comment (douglascamata) [inline cmd/rhelemeter-server/main_e2e_test.go:166]: Small suggestion here: might want to use `t.Cleanup(receiveServer.Close)`, as a `defer` won't run panic, but the first will.

**Review by douglascamata — APPROVED** [review]
Just a very small and non-blocking suggestion, LGTM! 🚀

**Comment by philipgough** [comment]
Thanks to you both for reviews. I need to add jsonnet changes assuming this all works in the end so will take care of both comments in said follow up.

**Review by moadz — APPROVED** [review]
LGTM

**Pull request merged**